projects/packages/password-checker/src/class-password-checker.php

```diff
@@ -0,0 +1,1313 @@
+<?php
+/**
+ * Password Checker package.
+ *
+ * @package  automattic/jetpack-password-checker
+ */
+
+namespace Automattic\Jetpack;
+
+/**
+ * Checks passwords strength.
+ */
+class Password_Checker {
+	/**
+	 * Minimum entropy bits a password should contain. 36 bits of entropy is considered
+	 * to be a reasonable password, 28 stands for a weak one.
+	 *
+	 * @var int
+	 */
+	public $minimum_entropy_bits = 28;
+
+	/**
+	 * Test results array.
+	 *
+	 * @var array
+	 */
+	public $test_results = array();
+
+	/**
+	 * Current password score.
+	 *
+	 * @var int
+	 */
+	public $score = 0;
+
+	/**
+	 * Current multiplier affecting the score.
+	 *
+	 * @var int
+	 */
+	public $multiplier = 4;
+
+	/**
+	 * A common password disallow list, which on match will immediately disqualify the password.
+	 *
+	 * @var array
+	 */
+	public $common_passwords = array();
+
+	/**
+	 * Minimum password length setting.
+	 *
+	 * @var int
+	 */
+	public $minimum_password_length = 6;
+
+	/**
+	 * User defined strings that passwords need to be tested for a match against.
+	 *
+	 * @var array
+	 */
+	private $user_strings_to_test = array();
+
+	/**
+	 * The user object for whom the password is being tested.
+	 *
+	 * @var mixed
+	 */
+	protected $user = null;
+
+	/**
+	 * The user identifier for whom the password is being tested, used if there's no user object.
+	 *
+	 * @var mixed
+	 */
+	protected $user_id = null;
+
+	/**
+	 * Creates an instance of the password checker class for the specified user, or
+	 * defaults to the currently logged in user.
+	 *
+	 * @param mixed $user can be an integer ID, or a WP_User object.
+	 */
+	public function __construct( $user = null ) {
+		if ( is_null( $user ) ) {
+			$this->user_id = get_current_user_id();
+		} elseif ( is_object( $user ) && isset( $user->ID ) ) {
+			// Existing user, using their ID.
+			$this->user_id = $user->ID;
+		} elseif ( is_object( $user ) ) {
+			// Newly created user, using existing data.
+			$this->user    = $user;
+			$this->user_id = 'new_user';
+		} else {
+			$this->user_id = $user;
+		}
+
+		/**
+		 * Filters the password strength enforcement settings.
+		 *
+		 * You can supply your own passwords that should not be used for authenticating in addition to weak and easy
+		 * to guess strings for each user. For example, you can add passwords from known password databases to avoid
+		 * compromised password usage.
+		 *
+		 * @param array $common_passwords strings that are forbidden for use as passwords.
+		 */
+		$this->common_passwords = apply_filters( 'password_checker_common_passwords', $this->common_passwords );
```

Comment:
To check for common passwords the test will need to use the password_checker_common_passwords filter to add strings to the array and then test to confirm using them as a password gives back a failure.

Reply:
Fixed.